### DISCUSSION TITLE
Functionality Overview

### DISCUSSION THREAD

**Discussion by david-clark-1043** [nss-day-cohort-54/c54-Rare-PP-client#62] [Ideas]
### Posts
| Issue | Title | Has been Commented | Has been coded | Done |
| - | - | - | - | - |
| #51 | View All Posts | :white_check_mark: | ✅  | :white_check_mark: |
| #50 | View Post Details | :white_check_mark: | :x: | :x: |
| #45 | Create Post | :white_check_mark: | :x: | :x: |
| #44 | View Current User's Posts |  ✅  | :x: | :x: |
| #43 | Edit A Post | :white_check_mark: | :x: | :x: |
| #42 | Delete A Post | :white_check_mark: | :x: | :x: |
| #41 | Filter Post List By Category | :x: | :x: | :x: |
| #39 | Filter By User | :x: | | |
| #29 | Filter Post List By Title | :white_check_mark: | :white_check_mark: | :white_check_mark: |
| #28 | Filter Post List By Tag | :x: | :x: | :x: |
| #34 | View Subscribed Posts | ✅  |  ✅  | :x: |

| Issue | Related Functionality | Client/Server | Comments | Code | Done |
| - | - | - | - | - | - |
| #51 | getAllPosts() | Client | :white_check_mark: | :white_check_mark: | :x: |
| #51 | get_all_posts() | Server | :white_check_mark: | :white_check_mark: | :x: |
| #39 | getPostsByUser() | Client | :white_check_mark: |  ✅  | :x: |
| #39 | get_posts_by_user() | Server | :white_check_mark: |  ✅  | :x: |
| #45 | create post func | Server | ✅ | ✅ |❌ |
| #42 | Delete A Post func | Server | ✅ | ✅ | :x: |
| #42 | Delete A Post component | Client | :x: | :x: | :x: |
| #43 | Edit A Post func | Server | ✅ | ✅ | :x: |
| #43 | Edit A Post component | Client | :x: | :x: | :x: |

### Categories
| Issue | Title | Has been Commented | Has been coded | Done |
| - | - | - | - | - |
| #49 | View Category List | :white_check_mark: | ✅  | :x: |
| #47 | Create A Category | :white_check_mark: | :x: | :x: |

| Issue | Related Functionality | Client/Server | Comments | Code | Done |
| - | - | - | - | - | - |
| #49 | getAllCategories() | Client | ✅  | ✅  | - |
| #49 | get_all_categories() | Server | ✅  | ✅  | - |
| #47 | createCategory() | Client | ✅  | - | - |
| #47 | create_category() | Server | ✅  | - | - |

~delete_category~ Commented, but not needed functionality

### Tags
| Issue | Title | Has been Commented | Has been coded | Done |
| - | - | - | - | - |
| #48 | View Tag List | :white_check_mark: | ✅  | :x: |
| #46 | Create A Tag | :white_check_mark: | ✅  | :x: |
| #38 | Add Tags to Post | :white_check_mark: | :x: | :x: |
| #37 | Update Tags on Post | :x: | :x: | :x: |

| Issue | Related Functionality | Client/Server | Comments | Code | Done |
| - | - | - | - | - | - |
| - | - | - | - | - | - |

### Comments
| Issue | Title | Has been Commented | Has been coded | Done |
| - | - | - | - | - |
| #33 | view Post's Comments | :white_check_mark: | :x: | :x: |
| #32 | Add comment to Post | :white_check_mark: | :x: | :x: |
| #31 | Delete a Comment | :white_check_mark: | :x: | :x: |

| Issue | Related Functionality | Client/Server | Comments | Code | Done |
| - | - | - | - | - | - |
| #33 | getCommentsByPostId() | Client | :white_check_mark: | - | - |
| #32 | addComment() | Client | :white_check_mark: | | | |
| #33 | CommentList Component | Client | :white_check_mark: | | | |
| #33 | Comment Component | Client | :white_check_mark: | | | |
| #32 | CommentForm Component | Client | :white_check_mark: | | | |
| #31 | deleteComment() | Client | :white_check_mark: | | | |
| #33 | get_comments_by_post() | Server | ✅ | | | |
| #32 | add_comment() | Server | ✅ | | | |
| #31 | delete_comment() | Server | ✅ | | | |

### Reactions
| Issue | Title | Has been Commented | Has been coded | Done |
| - | - | - | - | - |
| ? | :x: | :x: | :x: |

| Issue | Related Functionality | Client/Server | Comments | Code | Done |
| - | - | - | - | - | - |
| - | - | - | - | - | - |

### Subscriptions
| Issue | Title | Has been Commented | Has been coded | Done |
| - | - | - | - | - |
| #35 | Subscribe to a User | :white_check_mark: | ✅  | :x: |
| #30 | Unsubscribe from User | :white_check_mark: | ✅  | :x: |

| Issue | Related Functionality | Client/Server | Comments | Code | Done |
| - | - | - | - | - | - |
| #30 | deleteSubscription() | Client | :white_check_mark: | ✅ | :x: |
| #30 | delete_subscription() | Server | :white_check_mark: | ✅ | :x: |
| #35 | addSubscription() | Client | :white_check_mark: | ✅ | :x: |
| #35 | add_subscription() | Server | :white_check_mark: | ✅ | :x: |

### Users

| Issue | Title | Has been Commented | Has been coded | Done |
| - | - | - | - | - |
| #40 | View User List | :white_check_mark: | ✅ | :x: |
| #36 | View User Details | :white_check_mark: | ✅ | :x: |

| Issue | Related Functionality | Client/Server | Comments | Code | Done |
| - | - | - | - | - | - |
| #40 | getAllUsers() | Client | :white_check_mark: | :white_check_mark: | :x: |
| #40 | get_all_users() | Server | :white_check_mark: | ✅ | :x: |
| #36 | getSubscriptionsByUser() | Client | :white_check_mark: | ✅ | :x: |
| #36 | get_subscriptions_by_user() | Server | :white_check_mark: | ✅  | :x: |

**Comment by TomHawk123** [comment]
Where does this go?
